### PR TITLE
Add PDCurses to Windows

### PR DESCRIPTION
**Pull request by alef**
#### Summary
Build "Add PDCurses to Windows"

#### Purpose of change
Allow to build `TILES=0` for Windows with PDCurses from VCPKG.

#### Describe the solution
- Decouple `defined(_WIN32)` from `defined(TILES)` by using a new `#define TUI`
- Disable `wincurse.cpp`
- Add `USE_PDCURSES` to CMake builds
- Add more CMake presets

More details in the commits comments

#### Describe alternatives you've considered
None

#### Testing
Built with:
1. CMake Visual Studio presets
  - `windows-x64-msvc`
  - `windows-tiles-sounds-x64-msvc`
2. CMake WSL2 MSYS2 UCRT64 presets
  - `windows-x64`
  - `windows-tiles-sounds-x64`
3. CMake WSL2 Ubuntu presets
  - `linux-x64`
  - `linux-tiles-sounds-x64`
4. Makefile WSL2 MSYS2 UCRT64 with arguments
  - `CCACHE=1 RELEASE=1 MSYS=1 DYNAMIC_LINKING=1 SDL=0 TILES=0 SOUND=0 LOCALIZE=0 LINTJSON=0 ASTYLE=0 TESTS=0 cataclysm.exe`

Run inside Windows Terminal 1.19, open the main menu, load a debug game, open the overmap, quit:
* ~~MSVC has red and blue colors swapped. Can *not* be fixed with PDCurses' `-DPDC_RGB`.~~
* UCRT64 prints extra characters in some places
* MSYS' Makefile build crashes wit failed assertion in `imgui.cpp:9123` "Need a positive DeltaTime!"
* Console resizing doesn't work for now

Run SDL native or under WSLg.

#### Additional context
This PR in draft to check the CI build and the linters.

Related PR/issues:
- #27628
- #36034

This is the MSYS UCRT64 MINGW NCURSES build running in cmd.exe 80x25:
image

This is the MSVC PDCurses build running in cmd.exe 80x25:
image

### CODE REVIEW

**Comment by katemonster33** [comment]
Looks like this PR would fix #73011 which would be a huge weight off my shoulders when changes to ImTui need to be made

**Comment by alef** [comment]
Regarding _UCRT64 prints extra characters in some places_ I am now trying using `mingw-w64-ucrt-x86_64-pdcurses` instead of `mingw-w64-ucrt-x86_64-ncurses` as [suggested](https://github.com/ggerganov/imtui?tab=readme-ov-file#windows) by ImTui.

**Comment by katemonster33** [comment]
I get these statis assertion failures when building on Windows using VS2022 Community Edition:
```1>C:\Users\Kat\Documents\Cataclysm-DDA\src\ncurses_def.cpp(286,32): error C2338: static_assert failed: 'red must have the same value as COLOR_RED (from ncurses)'
1>C:\Users\Kat\Documents\Cataclysm-DDA\src\ncurses_def.cpp(290,35): error C2338: static_assert failed: 'yellow must have the same value as COLOR_YELLOW (from ncurses)'
1>C:\Users\Kat\Documents\Cataclysm-DDA\src\ncurses_def.cpp(292,33): error C2338: static_assert failed: 'blue must have the same value as COLOR_BLUE (from ncurses)'
1>C:\Users\Kat\Documents\Cataclysm-DDA\src\ncurses_def.cpp(296,33): error C2338: static_assert failed: 'cyan must have the same value as COLOR_CYAN (from ncurses)'
1>C:\Users\Kat\Documents\Cataclysm-DDA\src\ncurses_def.cpp(320,11): error C2039: 'is_term_resized': is not a member of '`global namespace''
1>C:\Users\Kat\Documents\Cataclysm-DDA\src\ncurses_def.cpp(320,11): error C3861: 'is_term_resized': identifier not found
1>C:\Users\Kat\Documents\Cataclysm-DDA\src\ncurses_def.cpp(338,47): error C2039: 'newscr': is not a member of '`global namespace''
1>C:\Users\Kat\Documents\Cataclysm-DDA\src\ncurses_def.cpp(338,43): error C2440: '<function-style-cast>': cannot convert from 'initializer list' to 'std::shared_ptr<void>'
1>C:\Users\Kat\Documents\Cataclysm-DDA\src\ncurses_def.cpp(349,5): error C3861: 'set_escdelay': identifier not found
1>C:\Users\Kat\Documents\Cataclysm-DDA\src\ncurses_def.cpp(437,17): error C2660: 'getmouse': function does not take 1 arguments```

File: msvc-full-features/Cataclysm-common.props [hunk 1]
```diff
@@ -52,24 +52,29 @@
       <AdditionalOptions>/bigobj /utf-8 %(AdditionalOptions)</AdditionalOptions>
       <LanguageStandard>stdcpp17</LanguageStandard>
       <AdditionalIncludeDirectories>$(MSBuildThisFileDirectory)..\pch;$(MSBuildThisFileDirectory)..\src;$(MSBuildThisFileDirectory)..\src\third-party;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
-      <PrecompiledHeader>Use</PrecompiledHeader>
-      <ForcedIncludeFiles>$(MSBuildThisFileDirectory)..\pch\main-pch.hpp</ForcedIncludeFiles>
       <DisableSpecificWarnings>4661;4819;4146;26495;26444;26451;4068;6319;6237</DisableSpecificWarnings>
-      <PreprocessorDefinitions>_SCL_SECURE_NO_WARNINGS;_CRT_SECURE_NO_WARNINGS;WIN32_LEAN_AND_MEAN;LOCALIZE;USE_VCPKG;IMGUI_DISABLE_OBSOLETE_KEYIO;%(PreprocessorDefinitions)</PreprocessorDefinitions>
```
Comment (akrieger) [inline msvc-full-features/Cataclysm-common.props:58]: @katemonster33 IDK what this does but I had to remove it for the notiles build to work. Can I make it unconditionally gone or do I need to keep this in tiles builds?

File: msvc-full-features/Cataclysm-vcpkg-static.vcxproj [hunk 2]
```diff
@@ -156,13 +157,20 @@
     <EmbedManifest>false</EmbedManifest>
     <TargetName>cataclysm-tiles</TargetName>
```
Comment (akrieger) [inline msvc-full-features/Cataclysm-vcpkg-static.vcxproj:158]: Probably could change the name of this in notiles also, but don't have energy to spin it back up right now.

**Comment by Maleclypse** [comment]
Is this ready or waiting on akrieger or something else?

**Comment by akrieger** [comment]
I made the non-curses build work but I don't have time to validate the curses build still works given the changes I had to make that I called out.

**Comment by alef** [comment]
[This](https://github.com/CleverRaven/Cataclysm-DDA/blame/master/src/third-party/imtui/imtui-impl-ncurses.cpp#L41) caused crashes in Windows debug mode because the debug RTL injects own things and `memset` destroy them.

**Pull request merged**